### PR TITLE
Use `list_item` for the component list in `InstancePath::data_ui`

### LINKED ISSUE #4161: Use the `ListItem` style for the component list (in the Selection Panel when an entity is selected)
This is a list with "synchronised" hover highlight just like the Blueprint and Streams UI. It should be styled the same way.

Design (ignore the fact that components are hierarchical here):
image
<br/><br/>

blocked by:
- https://github.com/rerun-io/rerun/issues/6245

### PR DESCRIPTION
**Pull request by abey79**
### What

- Fixes #4161
- Follow up to #6297
- Follow up to #6325

This PR uses `list_item` for the component list in the entity path selection panel and tooltip.

~~Blocked by https://github.com/emilk/egui/issues/4471 to resolve the tooltip "first frame flicker"~~

TODO:
-  [x] ⚠️ full_span_scope panic on hover on spatial space view: https://github.com/rerun-io/rerun/issues/6246

#### selection panel

before:

image

after:

image

*Note*: the hover behaviour is the biggest change here: now full span, previously just on the left-column button

#### tooltip

before:

image

after:
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6309?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6309?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6309)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by Wumpf** [comment]
before screenshots would be nice. my memory is pretty bad with these things :)

**Comment by abey79** [comment]
> before screenshots would be nice. my memory is pretty bad with these things :)

Done (though visually it doesn't make such a difference on screenshots)

File: crates/re_data_ui/src/instance_path.rs [hunk 1]
```diff
@@ -27,52 +24,88 @@ impl DataUi for InstancePath {
...
             if ctx.recording().is_known_entity(entity_path) {
                 // This is fine - e.g. we're looking at `/world` and the user has only logged to `/world/car`.
-                ui.label(format!(
-                    "No components logged on timeline {:?}",
-                    query.timeline().name()
-                ));
+                label_for_ui_layout(
+                    ui,
+                    ui_layout,
+                    format!(
```
Comment (emilk) [inline crates/re_data_ui/src/instance_path.rs:30]: could this be a member method?

```suggestion
                ui_layout.label(
                    ui,
                    format!(
```

**Review by emilk — APPROVED** [review]

File: crates/re_data_ui/src/instance_path.rs (hunk 1 above)
Comment (abey79) [reply crates/re_data_ui/src/instance_path.rs:30]: Actually, both this one and `table_for_ui_layout` could be methods. I like that.

**Pull request merged**